src/main/java/net/countercraft/movecraft/combat/movecraftcombat/listener/PistonListener.java

```diff
@@ -0,0 +1,69 @@
+package net.countercraft.movecraft.combat.movecraftcombat.listener;
+
+import net.countercraft.movecraft.combat.movecraftcombat.config.Config;
+import org.bukkit.Location;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockPistonRetractEvent;
+import org.bukkit.util.Vector;
+
+public class PistonListener implements Listener
+{
+    @EventHandler
+    public void onPistonRetract(BlockPistonRetractEvent e) {
+        if(!Config.ReImplementTranslocation)
+            return;
+
+        Location headLocation = e.getBlock().getLocation().add(blockFaceToVector(e.getDirection()));
+        Vector direction = blockFaceToVector(e.getDirection()).multiply(-1D);
+        Location moveLocation = e.getBlock().getLocation().add(direction);
+
+        //Block.isPassable() would be perfect for this but only exists in 1.13+
+        if(moveLocation.getBlock().getType().isSolid())
+            return;
+
+        for(Entity ent : e.getBlock().getWorld().getNearbyEntities(headLocation, 2D, 2D, 2D)) {
+            if(ent.getType() != EntityType.PRIMED_TNT)
+                continue;
+            if(!ent.getLocation().getBlock().getLocation().equals(headLocation) &&
```

Comment:
So many calls to getLocation().getBlock().getLocation(), store it above and check against it here.